references/validationFunctions/common/diseaseStatusFollowUp.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Enforces a required field if disease status at followup is progression or relapse
+ * @param {object} $row 
+ * @param {string} $field 
+ * @param {string} $name 
+ */
+const validation = ($row, $field, $name) => 
+    (function validate() {
+        let result = {valid: true, message: "Ok"};
+
+        /* required field, cannot be null */
+        const diseaseStatus = $row.disease_status_at_followup.trim().toLowerCase();
+
+        let isRequired = diseaseStatus === "relapse";
+
+        /* intended for distant / loco-regional progression, but would apply to any future progression types*/
+        const secondCondition = diseaseStatus.match(/(progression)$/);
+
+        const soloConditionFields = [
+            'relapse_interval'
+        ];
```

Comment:
If a field has a different requirement, shouldn't we just write a different validation script for that rather than complicating this one?

How about two scripts, one for `requiredWhenProgression` and `requiredWhenRelapse` and put both those scripts into a folder for followUp (since they are only used in the follow up file). Since we allow arrays of scripts now, one or both of these can be added in on the fields where needed.

Don't need the confusing `soloConditionFields` and `additionalMsg` logic then.

Reply:
should each individual field get its own personal 'combiner file' (the file that exports the individual scripts as one array) for clarity?

or, should fields that share the same combinations of scripts get a share a common 'combiner file'?

Reply:
felt like things would be more clear if "combiner" scripts were
- unique to every field. gives more flexibility for changes compared to making common combiners
- held separately from the scripts that they're composed of

I set it up like so as this made the most sense to me:
```
└── validationFunctions
    ├── follow_up
    │   ├── baseScripts
    │   │   ├── requiredWhenProgression.js
    │   │   └── requiredWhenRelapse.js
    │   ├── anatomic_site_progression_or_recurrences.js
    │   ├── method_of_progression_status.js
    │   ├── posttherapy_tumour_staging_system.js
    │   ├── recurrence_tumour_staging_system.js
    │   └── relapse_interval.js
```
changes made in: https://github.com/icgc-argo/argo-dictionary/pull/48/commits/97f90581f13b26653812f032d95752597172e185